feat: Implement terraform module with workspace and version

wdullaer opened the pull request:
#### Description
This PR implements a new module that can show a segment with the currently installed [Terraform](https://terraform.io) and workspace.

#### Motivation and Context
Showing the terraform workspace is a feature of spaceship that I missed when moving to starship.

I have considered a design where the package module is responsible for showing the currently selected workspace, but decided to isolate my changes in a new module for now.

Showing the terraform version is hidden behind a config flag and disabled by default. The `terraform version` command performs a lot of checks and gets very slow when more than 1 plugin is in use.
I have created PR hashicorp/terraform#23371 to remedy this.

Closes #522
Closes #532 (I hadn't seen this PR when I started working on this feature. This PR also adds terraform version support and acceptance tests.)

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):
(will come later)

#### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, tests ran to see how -->
<!--- your change affects other areas of the code, etc. -->
- [x] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

wdullaer commented:
The failing tests are due to the fact that terraform is not installed for the github actions.
I've been looking at fixing this, but haven't figured it out yet.

The documentation can use some updating here as well, since it still refers to azure pipelines.

Scotsguy commented on docs/de-DE/config/README.md:
````diff
@@ -845,6 +846,33 @@ The `rust` module shows the currently installed version of Rust. Das Modul wird
+[terraform]
+symbol = "🏎💨 "
+```
+
````
I don't think Crowdin likes pre-translated docs.
I assume that Crowdin will either overwrite this with the english translation, or still mark it untranslated.
Same with the other docs, it probably expects them to be untouched.

wdullaer replied:
I can easily revert back to the original text if that's preferred.

wdullaer replied:
I pushed new commits which remove the translations so they can be managed through crowdin

wdullaer commented:
Installing terraform with a github action in the way the tests expect is pretty big yak. I would like someone to confirm that the rest of the PR looks OK before I spend that effort.

wdullaer commented:
I refactored my code so it uses the new `utils::exec_cmd` helper where appropriate.

I also did a new search for setup-terraform github actions and found 2 candidates which did not exist when I originally made this PR. I added the one which I think looks most promising. This should normally make all the checks pass.

alexjurkiewicz commented:
This is very cool! Any news?

matchai commented on docs/config/README.md:
```diff
@@ -989,6 +990,33 @@ The module will be shown if any of the following conditions are met:
+| `symbol`       | `"💠 "`       | The symbol used before displaying the terraform workspace.  |
+| `style`        | `"bold 105"`  | The style for the module.                                   |
+| `disabled`     | `false`       | Disables the `terraform` module.                            |
+| `show_version` | `false`       | Shows the terraform version. Very slow on large workspaces. |
```
Nitpick: To keep the "Options" tables consistent, `style` and `disabled` are kept as the last two rows. Could we please move `show_version` to be above them? 😊

matchai commented on src/configs/terraform.rs:
```diff
@@ -0,0 +1,27 @@
+    pub version: SegmentConfig<'a>,
+    pub style: Style,
+    pub disabled: bool,
+    pub show_version: bool,
```
Similarly here ☝️

matchai commented on src/configs/terraform.rs:
```diff
@@ -0,0 +1,27 @@
+            version: SegmentConfig::default(),
+            style: Color::Fixed(105).bold(),
+            disabled: false,
+            show_version: false,
```
And here ☝️

matchai commented on src/modules/terraform.rs:
```diff
@@ -0,0 +1,125 @@
+        .set_extensions(&["tf"])
+        .is_match();
+
+    if !is_terraform_project {
```
👍

matchai commented on src/modules/terraform.rs:
```diff
@@ -0,0 +1,125 @@
+use std::io;
+use std::path::PathBuf;
+
+/// Creates a module with the current Terraform version and workspace
```
👍

matchai commented on src/modules/terraform.rs:
```diff
@@ -0,0 +1,125 @@
+    module.set_style(config.style);
+    module.create_segment("symbol", &config.symbol);
+
+    if config.show_version {
```
👍
